### PR TITLE
Credit Card expiryDate follow-up

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/5116.

### CODE REVIEW

File: server/graphql/v2/input/PaymentMethodInput.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export const getLegacyPaymentMethodFromPaymentMethodInput = async (
     };
     if (pm.creditCardInfo.expYear && pm.creditCardInfo.expMonth) {
       const { expYear, expMonth } = pm.creditCardInfo;
-      paymentMethod['expiryDate'] = moment.utc(`${expYear}-${expMonth}`, 'YYYY-MM');
+      paymentMethod['expiryDate'] = moment.utc(`${expYear}-${expMonth}`, 'YYYY-MM').endOf('month');
```
Comment (znarf) [inline server/graphql/v2/input/PaymentMethodInput.ts:87]: What about the one already added since the PR was deployed?

Comment (znarf) [reply server/graphql/v2/input/PaymentMethodInput.ts:87]: It's covered by the migration?

File: cron/monthly/expiring-cards.js [hunk 2]
```diff
@@ -20,13 +19,15 @@ if (config.env === 'production' && date !== 7 && date !== 21 && !process.env.OFF
...
     where: {
       type: 'creditcard',
-      data: {
-        expMonth: month,
-        expYear: year,
+      // Expiry Date is set to be the last second of the valid month.
+      // This query will return all the creditcards that will expire by the end of the current month.
+      expiryDate: {
+        [Op.gt]: moment.utc().startOf('month'),
+        [Op.lt]: moment.utc().endOf('month'),
```
Comment (znarf) [inline cron/monthly/expiring-cards.js:30]: What are the precise values being used in the comparaison practically?

Could it be that we end up with a condition like expiryDate < `2022-01-31 23:59:59` and expiryDate=2022-01-31 23:59:59. Not returning entries.

Comment (kewitz) [reply cron/monthly/expiring-cards.js:30]: Moment will return the timestamp to the last millisecond, but I'll update to `gte` and `lte` to be more explicit.

File: server/graphql/v2/input/PaymentMethodInput.ts (hunk 1 above)
Comment (kewitz) [reply server/graphql/v2/input/PaymentMethodInput.ts:87]: Yes, the migration will update existing credit cards.

**Review by Betree — APPROVED** [review]
The migration worked well on my local data. The code looks good for the rest

**Pull request merged**